scalahost-sbt/src/main/scala/scala/meta/scalahost/sbt/ScalahostSbtPlugin.scala

```diff
@@ -46,21 +59,75 @@ object ScalahostSbtPlugin extends AutoPlugin {
               )
             }
           List(
-            s"-Xplugin:${scalahostJar}",
+            s"-Xplugin:$scalahostJar",
             "-Yrangepos",
             "-Xplugin-require:scalahost"
           )
         case _ => Nil
       }
+    }
+  )
+
+  lazy val scalahostHasMirrorSettings: Seq[Def.Setting[_]] = Def.settings(
+    // Automatically set scalahostAggregate to projects listed in dependsOn(x % Scalahost)
+    scalahostAggregate := thisProject.value.dependencies.collect {
+      case x if x.configuration.exists(_ == Scalahost.name) => x.project
+    },
+    javaOptions ++= {
+      if (scalahostAggregate.value.isEmpty) Nil
+      else {
+        val sourcepath =
+          scalahostSourcepath.value
+            .map(_.getAbsolutePath)
+            .mkString(",")
+        val classpath =
+          scalahostClasspath.value
+            .flatMap(_.files.map(_.getAbsolutePath))
+            .mkString(java.io.File.pathSeparator)
+        List(
+          s"-Dscalameta.mirror.sourcepath=$sourcepath",
+          s"-Dscalameta.mirror.classpath=$classpath"
```

Comment:
The sys.props are unique per jvm process, so two independent projects won't clash. The only situation where this scheme fails if project A and project B both have `scalametaDependencies` and A depends on B (or vice-versa). I'm willing to bet this situation is very rare.

Using `-D<projectname>.scalameta.sourcepath` complicates the `Mirror()` constructor, since we don't know the project name.

How about doing both as a compromise? That is, `-D<projectname>.scalameta.sourcepath` and `-Dscalameta.sourcepath`